### ISSUE TITLE
Subcommands only display option groupings of its own options and the option of its parent command

### ISSUE THREAD

**Issue by Linard11** [yargs/yargs#1173]
If calling a subcommand with `--help`/`help`, it will only print groupings that were defined in the command itself, or the parent-command, but ignoring any groupings that were made in (anchestor-)commands further up the nesting, or in the global context:

**MCVE:**
`test.js`

```
const yargs = require('yargs');

yargs.option('foo').group(['foo'], 'Foo-Flags:');

yargs.command({
    command: 'com',
    describe: 'The Command',
    builder: yargs => {
        yargs.option('bar').group('bar', 'Bar-Flag:');

        yargs.command({
            command: 'subcom',
            describe: 'The Subcommand',
            builder: yargs => {
                yargs.option('baz').group('baz', 'Baz-Flag:');

                yargs.command({
                    command: 'subsubcom',
                    describe: 'The Subsubcommand',
                    builder: yargs => {
                        yargs.option('quax').group('quax', 'Quax-Flag:');
                    }
                });
            }
        });
    }
});

yargs.parse();
```

**Expected behavior:**
```
node test.js com subcom subsubcom --help

The Subsubcommand

Quax-Flag:
  --quax

Baz-Flag:
  --baz

Bar-Flag:
  --bar

Foo-Flags:
  --foo

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
```

**Actual behavior:**
```
node test.js com subcom subsubcom --help

The Subsubcommand

Quax-Flag:
  --quax

Baz-Flag:
  --baz

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --foo
  --bar
```

**Comment by cspotcode** [comment]
I am hitting the same problem.

I have global options placed in a group.

When I run `my-command --help` or `my-command foo --help` I see the groupings.

When I run `my-command foo bar --help` I do *not* see the groupings.

Going 2 levels deep triggers this bug.

**Comment by bcoe** [comment]
@Linard11 @cspotcode, I believe this should have been addressed in `yargs@14.2.0`. Let us know if you continue to bump into any issues.

Thanks for your patience 😄, and thanks for the fix @mleguen.
